src/main/java/org/prebid/server/auction/ExchangeService.java

```diff
@@ -950,6 +957,22 @@ private static void tryAddWinningBidByBidder(Bid bid, String bidder,
         return result;
     }
 
+    /**
+     * Creates {@link CacheBid} from given {@link Bid} and determined cache ttl.
+     */
+    private CacheBid toCacheBid(Bid bid, Map<String, Integer> impIdToTtl, BidRequestCacheInfo cacheInfo,
```

Comment:
Cache TTL calculation seems to me worthy of a separate service taking into account that it will be backed by DB in future.